services/snapshots/service.go

```diff
@@ -255,6 +255,29 @@ func (s *service) Usage(ctx context.Context, ur *snapshotsapi.UsageRequest) (*sn
 	return fromUsage(usage), nil
 }
 
+type cleaner interface {
+	Cleanup(ctx context.Context) error
+}
+
+func (s *service) Cleanup(ctx context.Context, cr *snapshotsapi.CleanupRequest) (*ptypes.Empty, error) {
+	sn, err := s.getSnapshotter(cr.Snapshotter)
+	if err != nil {
+		return nil, err
+	}
+
+	c, ok := sn.(cleaner)
+	if !ok {
+		return nil, errdefs.ToGRPCf(errdefs.ErrNotImplemented, "snapshotter does not implement Cleanup method")
+	}
+
+	err = c.Cleanup(ctx)
```

Comment:
I think we need to add implementation in metadata level.

Reply:
As a pass-through?

Reply:
just realize that the snapshotter interface is embedding in metadata snapshotter. no need to add implementation.